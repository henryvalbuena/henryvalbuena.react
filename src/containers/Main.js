import React from "react";
import { NavLink } from "react-router-dom";

import styles from "./Main.module.css";

const Main = () => (
  <div className={styles["main"]}>
    <div className={styles["main-context"]}>
      <h1>Do you really want to know?</h1>
      <h1>Choose</h1>
      <div className={styles["pills"]}>
        <NavLink to="/portfolio/red">
          <div className={styles["pills-container"]}>
            <i className={[styles["pill-red"], "pills icon"].join(" ")}></i>
            <span>Red Pill</span>
          </div>
        </NavLink>
        <NavLink to="/portfolio/blue">
          <div className={styles["pills-container"]}>
            <i className={[styles["pill-blue"], "pills icon"].join(" ")}></i>
            <span>Blue Pill</span>
          </div>
        </NavLink>
      </div>
    </div>
  </div>
);

export default Main;
