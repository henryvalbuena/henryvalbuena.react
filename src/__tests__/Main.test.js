import React from "react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import { render } from "@testing-library/react";
import Main from "../containers/Main";

describe("Renders", () => {
  it("Main renders", () => {
    const history = createMemoryHistory();
    const { getByText } = render(
      <Router history={history}>
        <Main />
      </Router>
    );
    expect(getByText("Do you really want to know?")).toBeInTheDocument();
  });

  it("Red pill", () => {
    const history = createMemoryHistory();
    const { getByText } = render(
      <Router history={history}>
        <Main />
      </Router>
    );
    expect(getByText("Red Pill")).toBeInTheDocument();
  });

  it("Blue pill", () => {
    const history = createMemoryHistory();
    const { getByText } = render(
      <Router history={history}>
        <Main />
      </Router>
    );
    expect(getByText("Blue Pill")).toBeInTheDocument();
  });
});

describe("Not ready (lazy render)", () => {
  it("Red lazy rendered (not present)", () => {
    const history = createMemoryHistory();
    const { queryByText } = render(
      <Router history={history}>
        <Main />
      </Router>
    );
    expect(queryByText("Projects")).toBeNull();
  });
  it("Blue lazy rendered (not present)", () => {
    const history = createMemoryHistory();
    const { queryByText } = render(
      <Router history={history}>
        <Main />
      </Router>
    );
    expect(queryByText("A little about me")).toBeNull();
  });
});

it("Snapshot App", () => {
  const history = createMemoryHistory();
  expect(
    render(
      <Router history={history}>
        <Main />
      </Router>
    )
  ).toMatchSnapshot();
});
