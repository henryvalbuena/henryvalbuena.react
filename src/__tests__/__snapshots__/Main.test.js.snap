// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot App 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main"
      >
        <div
          class="main-context"
        >
          <h1>
            Do you really want to know?
          </h1>
          <h1>
            Choose
          </h1>
          <div
            class="pills"
          >
            <a
              href="/portfolio/red"
            >
              <div
                class="pills-container"
              >
                <i
                  class="pill-red pills icon"
                />
                <span>
                  Red Pill
                </span>
              </div>
            </a>
            <a
              href="/portfolio/blue"
            >
              <div
                class="pills-container"
              >
                <i
                  class="pill-blue pills icon"
                />
                <span>
                  Blue Pill
                </span>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main"
    >
      <div
        class="main-context"
      >
        <h1>
          Do you really want to know?
        </h1>
        <h1>
          Choose
        </h1>
        <div
          class="pills"
        >
          <a
            href="/portfolio/red"
          >
            <div
              class="pills-container"
            >
              <i
                class="pill-red pills icon"
              />
              <span>
                Red Pill
              </span>
            </div>
          </a>
          <a
            href="/portfolio/blue"
          >
            <div
              class="pills-container"
            >
              <i
                class="pill-blue pills icon"
              />
              <span>
                Blue Pill
              </span>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
